framework/src/actions/SetupMeshCompleteAction.C

```diff
@@ -117,7 +117,8 @@ SetupMeshCompleteAction::act()
     // We currently only trigger the needsRemoteDeletion flag if somebody has requested a late
     // geometric ghosting functor and/or we have a displaced mesh. In other words, we almost never
     // trigger this.
-    if (_mesh->needsRemoteElemDeletion())
+    // We delete elements only when users ask us to do so AND libMesh allows us to do so
+    if (_mesh->needsRemoteElemDeletion() && _mesh->getMesh().allow_remote_element_removal())
```

Comment:
If you do this, then I don't think we will ever delete remote elements, even if we can. I think typically we set `allow_remote_element_removal = false` when there are late geometric ghosting functors, and then we have this code in `Action.C`:
```c++
  // Ok the above block may have told the mesh not to allow remote element removal during the
  // initial MeshBase::prepare_for_use, which is called after attaching geometric ghosting
  // functors. If we did tell the mesh not to allow remote element removal **and** we're using a
  // DistributedMesh, then we need to tell the mesh to allow remote element removal and ensure
  // that the mesh will delete its remote elements after the EquationSystems init
  else if (input_rm_type != Moose::RelationshipManagerType::GEOMETRIC &&
           (rm_type & Moose::RelationshipManagerType::GEOMETRIC) ==
               Moose::RelationshipManagerType::GEOMETRIC &&
           !rm_params.get<bool>("attach_geometric_early") && _mesh->isDistributedMesh())
  {
    _mesh->needsRemoteElemDeletion(true);
    if (_displaced_mesh)
      _displaced_mesh->needsRemoteElemDeletion(true);
  }
```
I did not add the code here, because I knew that I was going to end up deleting, but this code should also have `_mesh->getMesh().allow_remote_element_removal(true)` because at this point we are adding the late geometric ghosting functors and so it should be safe to delete the remote elements. So if you add this line, then you should go to this code in `Action.C` and add `_mesh->getMesh().allow_remote_element_removal(true);` in the same place that I am marking `needsRemoteElementDeletion(true)`.

If we have a problem with elements that should be ghosted getting deleted, then the fix is not to never delete remote elements, but instead to make sure we have the proper ghosting.

Reply:
OK, my point is that: if do ` mesh->getMesh().allow_remote_element_removal(false)` in my mesh generator, then I would like to keep all remotes regardless of RMs.

Reply:
I like something like: `force_no_remote_removal ` :-)

Reply:
Yikes yea that gets a bit dicey. In the code I pasted above we want to reenable remote element deletion *if* remote element deletion was disabled by the `Action`, but *not* if remote element deletion was disabled by a user's mesh generator.

I'm curious why you never want to allow remote element removal?

Reply:
At that point why use a `DistributedMesh`?

Reply:
Do you do some initial remote element deletion? Or are you reading in a split mesh?

Reply:
I am in distributed mesh generator ....

Reply:
> I'm curious why you never want to allow remote element removal?

I'm still curious about the answer to this question.

Reply:
I will add RMs so that we can get rid of this fragile parameter.

Reply:
Now that you are serializing the displaced mesh for exodus output, do you still need this change? What tests fail if you don't have it? I am worried that we will forget to remove this code in the future and a lot of our meshes will be serial without even realizing it.

Additionally, I am curious what tests fail if you don't have your `allow_remote_element_removal(false)` in your `DistributedMeshGenerator`. I know that you put it there for a reason...something to do with periodic boundary conditions I believe.

Reply:
> Now that you are serializing the displaced mesh for exodus output, do you still need this change? What tests fail if you don't have it? I am worried that we will forget to remove this code in the future and a lot of our meshes will be serial without even realizing it.

Adaptivity of mechanics problems, in general, won't work on a `distributed` mode without the changes. I GUESS we do not have these problems for regular users possibly because tensor mechanics most of the time stay with replicated mesh

Reply:
> Additionally, I am curious what tests fail if you don't have your `allow_remote_element_removal(false)` in your `DistributedMeshGenerator`. I know that you put it there for a reason...something to do with periodic boundary conditions I believe.

Periodic boundary tests will fail. The reason is that:

 After I have done mesh generation, a `prepare_for_use` will be involved inside of mesh generator, but the mesh generator has no idea whether periodic BCs will be used later.  Libmesh will remove elements that are needed for periodic BCs later.

Reply:
The fix would happen if we know we should keep for the later simulation. It has to be implemented by running `add_rm` before `mesh generation`.

Unfortunately, `ElementSideNeighborLayers` required for periodic BCs has to know mesh information ... but I do not have that information before mesh generation done.

Reply:
I do not need `_mesh->getMesh().allow_remote_element_removal()` anymore once RMs are setup correctly.